### DISCUSSION TITLE
Why can't I download the file?

### DISCUSSION THREAD

**Discussion by Jorge-S13** [nutgram/nutgram#494] [Q&A]
I am trying to download a file. here is an example from the documentation. but for some reason it doesn't work.
```php
$fileId = $bot->message()->sticker->file_id;
$bot->getFile($fileId)->save('storage/app/public');
```

I am trying to download a photo of my bot. but it doesn't work. Here is my code
```php
$botPhoto = $bot->getUserProfilePhotos(['user_id' => $bot->getMe()->id])->photos;
$photo = $botPhoto[0][2];
$fileId = $photo->file_id;
$file = $bot->getFile($fileId);
$bot->downloadFile($file, public_path('app/public'));
```
My app/public folder is always empty.

**Comment by Lukasss93** [comment]
any errors/logs?

**Reply by Jorge-S13** [reply]
No errors. Here are my logs when I send a sticker to chat
`[2023-06-16 12:35:46] DEBUG: getUpdates
{"ok":true,"result":[{"update_id":776725314,
"message":{"message_id":914,"from":{"id":3182829,"is_bot":false,"first_name":"Jorge","username":"sviridov13","language_code":"ru"},"chat":{"id":3182829,"first_name":"Jorge","username":
"sviridov13","type":"private"},"date":1686919012,"sticker":{"width":512,"height":512,"emoji":"\ud83d\udc4d","set_name":"RickAndMorty","is_animated":true,"is_video":false,"type":"regula
r","thumbnail":{"file_id":"AAMCAgADGQEAAgOSZIxXZDwl_dRpK5AAAWUZTveaPYqXAAImAwACtXHaBj4ZC4vnHBlAAQAHbQADLwQ","file_unique_id":"AQADJgMAArVx2gZy","file_size":5174,"width":128,"height":12
8},"thumb":{"file_id":"AAMCAgADGQEAAgOSZIxXZDwl_dRpK5AAAWUZTveaPYqXAAImAwACtXHaBj4ZC4vnHBlAAQAHbQADLwQ","file_unique_id":"AQADJgMAArVx2gZy","file_size":5174,"width":128,"height":128},"file_id":"CAACAgIAAxkBAAIDkmSMV2Q8Jf3UaSuQAAFlGU73mj2KlwACJgMAArVx2gY-GQuL5xwZQC8E","file_unique_id":"AgADJgMAArVx2gY","file_size":18212}}}]}
{"parameters":{"offset":776725314,"limit":100,"timeout":"10"},"options":{"timeout":11}}
`

**Reply by Jorge-S13** [reply]
Could it be a cache issue?

**Reply by Lukasss93** [reply]
I don't think so

**Reply by Lukasss93** [reply]
Nutgram version? Framework used?

**Reply by Jorge-S13** [reply]
nutgram 3 and laravel 9
this is what i got in the logs
`[2023-06-16 12:51:55] DEBUG: getUserProfilePhotos
{"ok":true,"result":{"total_count":1,"photos":[[{"file_id":"AgACAgIAAxUAAWSMSOxC51wtFMUo_juQ7-tI40rgAALYyDEbU5lwS_Puo4WsbS_VAQADAgADYQADLwQ","file_unique_id":"AQAD2MgxG1OZcEsAAQ","file
_size":12410,"width":160,"height":160},{"file_id":"AgACAgIAAxUAAWSMSOxC51wtFMUo_juQ7-tI40rgAALYyDEbU5lwS_Puo4WsbS_VAQADAgADYgADLwQ","file_unique_id":"AQAD2MgxG1OZcEtn","file_size":3385
5,"width":320,"height":320},{"file_id":"AgACAgIAAxUAAWSMSOxC51wtFMUo_juQ7-tI40rgAALYyDEbU5lwS_Puo4WsbS_VAQADAgADYwADLwQ","file_unique_id":"AQAD2MgxG1OZcEsB","file_size":86339,"width":640,"height":640}]]}}
{"parameters":{"user_id":5948028649},"options":[]}

[2023-06-16 12:51:55] DEBUG: getFile
{"ok":true,"result":{"file_id":"AgACAgIAAxUAAWSMSOxC51wtFMUo_juQ7-tI40rgAALYyDEbU5lwS_Puo4WsbS_VAQADAgADYwADLwQ","file_unique_id":"AQAD2MgxG1OZcEsB","file_size":86339,"file_path":"photos/file_0.jpg"}}
{"parameters":{"file_id":"AgACAgIAAxUAAWSMSOxC51wtFMUo_juQ7-tI40rgAALYyDEbU5lwS_Puo4WsbS_VAQADAgADYwADLwQ"},"options":[]}
`

**Reply by Jorge-S13** [reply]
but my folder is empty

**Reply by sergix44** [reply]
Please make sure that the user thats running the bot has write permission to that folder. Whats the return value of the save or downloadFIle functions?

**Reply by Jorge-S13** [reply]
The file appears to have downloaded. it appeared in my folder. but I don't know his formula. I can't open it. i am using windows. in what format should the downloaded image be saved?

**Reply by Lukasss93** [reply]
https://github.com/nutgram/nutgram/blob/master/src/Telegram/Types/Media/File.php#L49

**Reply by Jorge-S13** [reply]
It looks like this https://imgur.com/a/6ck2vUE

**Reply by Jorge-S13** [reply]
Sorry for stupid questions. Everything is working
